R/teal_lockfile.R

```diff
@@ -0,0 +1,90 @@
+#' Create `renv` `.lockfile`
+#'
+#' This function is used during `teal::init` and enables to create `renv` `.lockfile` to be shared in the `teal` app.
+#'
+#' The process uses [renv::snapshot()], which allows multiple ways of `.lockfile` creation.
+#'
+#' - **Pre-computed `.lockfile`**: User is able to provide his own pre-computed `.lockfile` by setting the path to the
+#' `.lockfile` through `options(teal.renv.lockfile = "")`. Then this function is not used.
+#' - **Working directory `.lockfile`**: If `options(teal.renv.lockfile)` is empty, by default `teal` will
+#' create an `implicit` type of the `.lockfile`, that uses `renv::dependencies()` to detect all R packages in the
+#' current project working directory.
+#' - **`DESCRIPTION` based `.lockfile`**: You can always include a `DESCRIPTION` file in your working directory and
+#' enable `.lockfile` creation based on this file. To do this, run `renv::settings$snapshot.type("explicit")`.
+#' Naming of `type` is the same as in `renv::snapshot()`. For the `"explicit"` type refer to
+#' `renv::settings$package.dependency.fields()`
+#' to see what `DESCRIPTION` fields are included in the `.lockfile`.
+#' - **Custom files based `.lockfile`**: If you want to specify custom files as a base for the `.lockfile`, then run
+#' `renv::settings$snapshot.type("custom")` and set `renv.snapshot.filter` option.
```

Comment:
This part needs rework.
I believe the decission tree is as follows:
- user specified lock file
- automatically computed lock file
  - based on working dir
  - based on DESC file
  - custom files

Reply:
on it

Reply:
<img width="536" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/7c125d5f-3c24-4a33-850d-3aa582d6df26">

fdffe3db7039e33be6c982760ffe439a2094cecb